### PR TITLE
feat: Introduce cloud installation actions for spawning mattermost cloud instances

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

Created reusable Github Actions to manipulate mattermost cloud instances
- Created install-mattermost-cloud action
- Created create-mattermost-cloud action
- Created delete-mattermost-cloud action

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-7291

### CODE REVIEW

File: delivery/create-cloud-installation/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  create-cloud-installation:
+    runs-on: <private-runners>
+    steps:
+      - uses: mattermost/actions/delivery/install-mattermost-cloud-@main
+        with:
+          version: 0.83.0
+          
+      - uses: mattermost/actions/delivery/create-cloud-installation@main
+        with:
+          dns: my-first-instance.test.mattermost.com
```
Comment (saturninoabril) [inline delivery/create-cloud-installation/README.md]: Should this be:
```suggestion
          dns: my-first-instance
```

`dns` sounds confusing atleast to me. Maybe `name` or `subdomain`?

File: delivery/create-cloud-installation/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          server: https://delivery-provisioner.test.mattermost.com
```
Comment (saturninoabril) [inline delivery/create-cloud-installation/README.md:19]: Is the idea for this action to cater different provisioner server depending on which repo will consume or we will use the same in all places? If the latter, should this be defined within the action itself and no need to expose together with its corresponding `x-api-key`?

File: delivery/create-cloud-installation/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            KEY_NAME2=VALUE2
+          headers: |-
+            x-api-key=MY_TOKEN
+```
+
+### Inputs
+
+| Input                       | Description                                                                                                                                                                                     | Default                                  | Required |
+| --------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------- | -------- |
+| dns                         | URLs at which the Mattermost server will be available.                                                                                                                                          |                                          | true     |
```
Comment (saturninoabril) [inline delivery/create-cloud-installation/README.md]: Same comment as above. This seemed to accept a name/subdomain but not URL.

File: delivery/delete-cloud-installation/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          headers: |-
+            x-api-key=MY_TOKEN
+```
+
+### Inputs
+
+| Input           | Description                                                                                        | Default | Required |
+| --------------- | -------------------------------------------------------------------------------------------------- | ------- | -------- |
+| server          | The provisioning server whose API will be queried.                                                 |         | true     |
+| headers         | The headers to add in every API call towards the provisioning server. Accepts new line list format |         | false    |
```
Comment (saturninoabril) [inline delivery/delete-cloud-installation/README.md:29]: Same comment here if server and header need to be exposed to consumer or just declare within the action itself.

**Review by saturninoabril — COMMENTED** [review]
Thanks @toninis and this is awesome! All good to me except for few clarifications.

File: delivery/create-cloud-installation/README.md (hunk 1 above)
Comment (toninis) [reply delivery/create-cloud-installation/README.md]: You are correct. I will switch that to `name` . It makes more sense

File: delivery/create-cloud-installation/README.md (hunk 2 above)
Comment (toninis) [reply delivery/create-cloud-installation/README.md:19]: This is an internal server in any case and the KEY will be a secret . In the future we might need to use different provisioner URLs so I do not want to hardcode it . 
We can keep the URL a secret too if that is the case, but again this is a public DNS name pointing on internal IPs

File: delivery/delete-cloud-installation/README.md (hunk 4 above)
Comment (toninis) [reply delivery/delete-cloud-installation/README.md:29]: Yep it needs to be exposed . This is a reusable action and might be used in different places

**Review by saturninoabril — APPROVED** [review]

**Review by phoinixgrr — APPROVED** [review]
🚀

**Pull request merged**